source/common/upstream/cluster_manager_impl.cc

```diff
@@ -156,95 +156,48 @@ void ClusterManagerInitHelper::setInitializedCb(std::function<void()> callback)
   }
 }
 
-void ClusterManagerImpl::initializeClustersFromV1Json(const Json::Object& config) {
-  envoy::api::v2::ConfigSource cds_config;
-  if (config.hasObject("cds")) {
-    envoy::api::v2::Cluster cds_cluster;
-    Config::CdsJson::translateCluster(*config.getObject("cds")->getObject("cluster"), sds_config_,
-                                      cds_cluster);
-    loadCluster(cds_cluster, false);
-    Config::Utility::translateCdsConfig(*config.getObject("cds"), cds_config);
-    // We can now potentially create the CDS API once the backing cluster exists.
-    cds_api_ = factory_.createCds(cds_config, sds_config_, *this);
-    init_helper_.setCds(cds_api_.get());
-  } else {
-    init_helper_.setCds(nullptr);
-  }
-
-  for (const Json::ObjectSharedPtr& cluster_config : config.getObjectArray("clusters")) {
-    envoy::api::v2::Cluster cluster;
-    Config::CdsJson::translateCluster(*cluster_config, sds_config_, cluster);
-    loadCluster(cluster, false);
-  }
-}
-
-void ClusterManagerImpl::initializeClustersFromV2Proto(const envoy::api::v2::Bootstrap& bootstrap) {
-  for (const auto& cluster : bootstrap.static_resources().clusters()) {
-    loadCluster(cluster, false);
-  }
-
-  // We can now potentially create the CDS API once the backing cluster exists.
-  if (bootstrap.dynamic_resources().has_cds_config()) {
-    cds_api_ = factory_.createCds(bootstrap.dynamic_resources().cds_config(), sds_config_, *this);
-    init_helper_.setCds(cds_api_.get());
-  } else {
-    init_helper_.setCds(nullptr);
-  }
-}
-
-ClusterManagerImpl::ClusterManagerImpl(const Json::Object& config,
-                                       const envoy::api::v2::Bootstrap& bootstrap,
+ClusterManagerImpl::ClusterManagerImpl(const envoy::api::v2::Bootstrap& bootstrap,
                                        ClusterManagerFactory& factory, Stats::Store& stats,
                                        ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime,
                                        Runtime::RandomGenerator& random,
                                        const LocalInfo::LocalInfo& local_info,
                                        AccessLog::AccessLogManager& log_manager)
     : factory_(factory), runtime_(runtime), stats_(stats), tls_(tls.allocateSlot()),
       random_(random), local_info_(local_info), cm_stats_(generateStats(stats)) {
-
-  config.validateSchema(Json::Schema::CLUSTER_MANAGER_SCHEMA);
-
-  if (config.hasObject("outlier_detection")) {
-    std::string event_log_file_path =
-        config.getObject("outlier_detection")->getString("event_log_path", "");
+  const auto& cm_config = bootstrap.cluster_manager();
+  if (cm_config.has_outlier_detection()) {
+    std::string event_log_file_path = cm_config.outlier_detection().event_log_path();
```

Comment:
Is the reason this is not `const std::string&` string compat issues? (could at least make const)

Reply:
I think that was accidental, but now that you mention it, we probably should be doing that, so all good :)